docs/coordinates/frames.rst

```diff
@@ -282,71 +282,188 @@ documentation.
+allows specifying an ``equinox`` that helps define the transformation between
+`~astropy.coordinates.FK5` and the `~astropy.coordinates.ICRS` frame. Frame
+attributes are defined by creating class attributes that are instances of
+`~astropy.coordinates.Attribute` or its subclasses (e.g.,
+`~astropy.coordinates.TimeAttribute`, `~astropy.coordinates.QuantityAttribute`,
+etc.). If attributes are defined using these classes, there is often no need to
+define an ``__init__`` function, as the initializer in
+`~astropy.coordinates.BaseCoordinateFrame` will probably behave in the way you
+want. Let's now modify the above toy frame class implementation to add two frame
+attributes::
+
+    >>> from astropy.coordinates import TimeAttribute, QuantityAttribute
+    >>> class MyFrame4(BaseCoordinateFrame):
+    ...     # Specify how coordinate values are represented when outputted
+    ...     default_representation = r.SphericalRepresentation
+    ...
+    ...     # Override component names (e.g., "ra" instead of "lon")
+    ...     frame_specific_representation_info = {
+    ...         r.SphericalRepresentation: [RepresentationMapping('lon', 'R'),
+    ...                                     RepresentationMapping('lat', 'D')],
+    ...         r.CartesianRepresentation: [RepresentationMapping('x', 'a'),
+    ...                                     RepresentationMapping('y', 'b'),
+    ...                                     RepresentationMapping('z', 'c')]
+    ...     }
+    ...
+    ...     # Specify frame attributes required to fully specify the frame
+    ...     time = TimeAttribute(default='B1950')
+    ...     orientation = QuantityAttribute(default=42*u.deg)
+
+Without specifying an initializer, defining these attributes tells the
+`~astropy.coordinates.BaseCoordinateFrame` what to expect in terms of additional
+arguments passed in to our subclass initializer. For example, when defining a
+frame instance with our subclass, we can now optionally specify values for these
+attributes::
+
+    >>> fr = MyFrame4(R=1*u.deg, D=2*u.deg, orientation=21*u.deg)
+    >>> fr # doctest: +FLOAT_CMP
+    <MyFrame4 Coordinate (time=B1950.000, orientation=21.0 deg): (R, D) in deg
+        (1., 2.)>
+
+Note that we specified both frame attributes with default values, so they are
+optional arguments to the frame initializer. Note also that the frame attributes
+now appear in the ``repr`` of the frame instance above. As a bonus, for most of
+the ``Attribute`` subclasses, even without defining an initializer, attributes
+specified as arguments will be validated. For example, arguments passed in to
+`~astropy.coordinates.QuantityAttribute` attributes will be checked that they
+have valid and compatible units with the expected attribute units. Using our
+frame example above, which expects an ``orientation`` with angular units,
+passing in a time results in an error::
+
+    >>> MyFrame4(R=1*u.deg, D=2*u.deg, orientation=55*u.microyear) # doctest: +SKIP
```

Comment:
I think you'd want `# doctest: +IGNORE_EXCEPTION_DETAIL` (e.g., `time/index.rst` for examples)